functions/autopilotRedirect.protected.ts

```diff
@@ -1,40 +1,95 @@
-import { Context, ServerlessCallback } from '@twilio-labs/serverless-runtime-types/types';
-import { JsonObject } from 'swagger-ui-express';
+import {
+  Context,
+  ServerlessCallback,
+  ServerlessFunctionSignature,
+} from '@twilio-labs/serverless-runtime-types/types';
 
 export interface Event {
   Memory: string;
+  Channel: string;
+  UserIdentifier: string;
 }
 
-export const handler = (context: Context, event: Event, callback: ServerlessCallback) => {
+type EnvVars = {
+  TWILIO_CHAT_TRANSFER_WORKFLOW_SID: string;
+};
+
+const handleChatChannel = async (context: Context<EnvVars>, event: Event) => {
+  const memory = JSON.parse(event.Memory);
+  const { ServiceSid, ChannelSid } = memory.twilio.chat;
+
+  const channel = await context
+    .getTwilioClient()
+    .chat.services(ServiceSid)
+    .channels(ChannelSid)
+    .fetch();
+
+  const attributes = JSON.parse(channel.attributes);
+
+  // if channel is webchat, disable the input
+  if (attributes.channel_type === 'web') {
+    const user = await context
+      .getTwilioClient()
+      .chat.services(ServiceSid)
+      .users(event.UserIdentifier)
+      .fetch();
+
+    const userAttr = JSON.parse(user.attributes);
+    const updatedAttr = { ...userAttr, lockInput: true };
```

Comment:
This seems to lock the input in every situation, even if input is needed for the chatbot.  Currently on staging chat, you get stuck because it asks a question but the input is locked.  Is that an issue with this?

![image](https://user-images.githubusercontent.com/10714292/94500616-a8a41a00-01b4-11eb-8175-28c8499d65b8.png)

Reply:
That is the behavior of webchat with the actual version of staging: the input is hidden as soon as the user starts the chat (regardless of the presence of the chatbot), and it's re-enabled after a counselor joins the chat (which will never happen as we are stuck with the bot). To test the functionality in this PR, you must use the chat version in https://github.com/tech-matters/webchat/pull/9